### PR TITLE
Port statement import

### PR DESCRIPTION
**Pull request by steveferry**
Porting g11, v11, CSV (UBS, Raffaisen) parsers

### CODE REVIEW

**Comment by ecino** [comment]
You need to fix tests for module lsv_dd in :
l10n_ch_lsv_dd/test/lsv-dd-test.yml

You need to fix PEP8:
./l10n_ch_lsv_dd/model/bank.py:25:1: E302 expected 2 blank lines, found 1

Also, since module lsv_dd is taken out from **unported**, it would be nice to adapt the code to V8.

**Comment by ecino** [comment]
Can someone add the workinprogress label please ? We are not yet ready with this one.

**Comment by max3903** [comment]
CLA are fixed.

**Comment by ecino** [comment]
@guewen We are ready with this now, can you remove work in progress label ?

File: l10n_ch_account_statement_base_import/README.rst [hunk 1]
```diff
@@ -0,0 +1,70 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Module name
```
Comment (yvaucher) [inline l10n_ch_account_statement_base_import/README.rst]: Put the module name here

File: l10n_ch_account_statement_base_import/README.rst [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+
+Credits
+=======
+
+Contributors
+------------
+
+* Nicolas Bessi <nicolas.bessi@camptocamp.com>
+* Please Compassion put your email here.
```
Comment (yvaucher) [inline l10n_ch_account_statement_base_import/README.rst]: @ecino to add

Emanuel Cino ecino@compassion.ch

File: l10n_ch_account_statement_base_import/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+{'name': "Swiss bank statements import",
+ 'version': '0.4',
```
Comment (yvaucher) [inline l10n_ch_account_statement_base_import/__openerp__.py]: Please put full version `8.0.0.4.0`

File: l10n_ch_account_statement_base_import/account_bank_statement_import.py [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+                    attachment = self.env['ir.attachment'].create(
+                        {
+                            'name': attachment[0],
+                            'res_model': 'account.bank.statement.line',
+                            'res_id': statement_line.id,
+                            'type': 'binary',
+                            'datas': attachment[1],
+                        }
+                    )
+                    statement_line.related_file = [attachment.id]
```
Comment (yvaucher) [inline l10n_ch_account_statement_base_import/account_bank_statement_import.py]: attachment is a record set you can assign it directly

File: l10n_ch_account_statement_base_import/account_bank_statement_import.py [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+        :type data_file: string
+
+        :return: A generator of available parser instances
+        :rtype: generator of available parser instances
+
+        """
+        for parser_class in base_parser.BaseSwissParser.__subclasses__():
+            yield parser_class(data_file)
+
+    def _create_bank_statements(self, stmts_vals):
```
Comment (yvaucher) [inline l10n_ch_account_statement_base_import/account_bank_statement_import.py]: Please add `@api.model`

File: l10n_ch_account_statement_base_import/parser/g11_file_parser.py [hunk 6]
```diff
@@ -0,0 +1,230 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Author: Steve Ferry
```
Comment (yvaucher) [inline l10n_ch_account_statement_base_import/parser/g11_file_parser.py]: Can you add a year with copyright ?

**Comment by steveferry** [comment]
Fix made

**Comment by ecino** [comment]
Can you rebase on OCA/8.0 ? It should remove Travis errors.

**Comment by ecino** [comment]
You need to install xlrd for the tests by adding it to travis (with pip for instance, avoid using sudo)

**Comment by eLBati** [comment]
@steveferry please remove the ported modules from `__unported__` folder

File: .travis.yml [hunk 7]
```diff
@@ -30,7 +30,9 @@ virtualenv:
 install:
   - pip install -q unidecode
   - pip install unicodecsv
+  - pip install xlrd
   - git clone --depth=1 https://github.com/OCA/maintainer-quality-tools.git ${HOME}/maintainer-quality-tools
+  - git clone https://github.com/OCA/bank-statement-import.git ${HOME}/build/CompassionCH/bank-statement-import -b 8.0
```
Comment (yvaucher) [inline .travis.yml]: To add bank-statement-import, please add it in oca_dependencies.txt

https://github.com/OCA/l10n-switzerland/blob/7.0/oca_dependencies.txt

File: l10n_ch_account_statement_base_import/README.rst [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+set to statement currency.
+
+Usage
+=====
+
+To use this module, you need to:
+
+
+Known issues / Roadmap
+======================
```
Comment (yvaucher) [inline l10n_ch_account_statement_base_import/README.rst:40]: Please remove those empty sections

File: l10n_ch_account_statement_base_import/README.rst [hunk 9]
```diff
@@ -0,0 +1,70 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Swiss bank statements import
+===========
```
Comment (yvaucher) [inline l10n_ch_account_statement_base_import/README.rst]: Would be nice to complete this line.

**Comment by ecino** [comment]
Does someone know why the runbot fails on dependencies (travis is fine) ?

File: l10n_ch_account_statement_base_import/README.rst [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+
+Credits
+=======
+
+Contributors
+------------
+
+* Nicolas Bessi <nicolas.bessi@camptocamp.com>
+* Emanuel Cino <ecino@compassion.ch>
+
```
Comment (ecino) [inline l10n_ch_account_statement_base_import/README.rst:51]: @steveferry Add your name here

File: l10n_ch_account_statement_base_import/account_bank_statement_import.py [hunk 11]
```diff
@@ -0,0 +1,183 @@
...
+        comodel_name='ir.attachment',
+        string='Related files',
+        readonly=True
+    )
+
+
+class AccountBankStatementLine(models.Model):
+    _inherit = 'account.bank.statement.line'
+
+    related_file = fields.Many2many(
```
Comment (ecino) [inline l10n_ch_account_statement_base_import/account_bank_statement_import.py]: Can a line have several attachments ? If yes, maybe we could rename this field to `related_files`.
If not, maybe we could change to `Many2one` ?

File: l10n_ch_account_statement_base_import/view/statement_view.xml [hunk 12]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (ecino) [inline l10n_ch_account_statement_base_import/view/statement_view.xml]: This view is not needed anymore. It references old field (already_completed) which does not exist in v8 and adds a button (download_attachment) that has no corresponding code to be called. Please remove this file and its reference in openerp manifest.

**Comment by yvaucher** [comment]
Superseeded by #162

**Pull request closed without merging**